src/platform_impl/linux/x11/ime/context.rs

```diff
@@ -355,25 +354,44 @@ impl ImeContext {
         self.allowed
     }
 
-    // Set the spot for preedit text. Setting spot isn't working with libX11 when preedit callbacks
-    // are being used. Certain IMEs do show selection window, but it's placed in bottom left of the
-    // window and couldn't be changed.
-    //
-    // For me see: https://bugs.freedesktop.org/show_bug.cgi?id=1580.
-    pub(crate) fn set_spot(&mut self, xconn: &Arc<XConnection>, x: c_short, y: c_short) {
-        if !self.is_allowed() || self.ic_spot.x == x && self.ic_spot.y == y {
+    /// Set the spot and area for preedit text.
+    ///
+    /// This functionality depends on the libx11 version.
+    ///  - Until libx11 1.8.2, XNSpotLocation was blocked by libx11 in On-The-Spot mode.
+    ///  - Until libx11 1.8.11, XNArea was blocked by libx11 in On-The-Spot mode.
+    ///
+    /// Use of this information is discretionary by input method servers,
+    /// and some may not use it by default, even if they have support.
+    pub(crate) fn set_area(
+        &mut self,
+        xconn: &Arc<XConnection>,
+        x: i16,
+        y: i16,
+        width: u16,
+        height: u16,
+    ) {
+        if !self.is_allowed()
+            || self.ic_area.x == x
+                && self.ic_area.y == y
+                && self.ic_area.width == width
+                && self.ic_area.height == height
+        {
             return;
         }
 
-        self.ic_spot = ffi::XPoint { x, y };
+        self.ic_area = ffi::XRectangle { x, y, width, height };
```

Comment:
Can you construct new area before `if` and compare the struct?

Reply:
Fair enough. Given how it was done previously I wasn't sure initially whether these ffi structs supported comparison.